src/coreclr/src/classlibnative/bcltype/system.cpp

```diff
@@ -607,9 +607,28 @@ BOOL QCALLTYPE SystemNative::WinRTSupported()
 
 #endif // FEATURE_COMINTEROP
 
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
 
+void QCALLTYPE SystemNative::X86BaseCpuId(int cpuInfo[4], int functionId, int subFunctionId)
+{
+    QCALL_CONTRACT;
 
+    BEGIN_QCALL;
 
+#if defined(_MSC_VER)
```

Comment:
You can avoid this ifdef by renaming `getextcpuid` to `__cpuidex`, and deleting implementation of `getextcpuid` on Windows. The less handwritten assembly we have, the better.

Reply:
Can do, but it will also require reordering the parameters on Unix; as it takes them in a completely different order.

Also, is there any reason why we have it as actual assembly rather than just copying `cpuid.h` from clang (Apache License v2.0 with LLVM Exceptions): https://github.com/llvm/llvm-project/blob/master/clang/lib/Headers/cpuid.h? I'd guess maybe just legacy?

The implementation they have uses `__asm`, which will produce better and less error prone codegen. `macOS` also has it, but appears to need a reference to Kernel.Frameworks for it to be accessible (at least from some brief searching)

Reply:
>  it will also require reordering the parameters on Unix; as it takes them in a completely different order.

I think that's fine.

> is there any reason why we have it as actual assembly rather than just copying cpuid.h

This `cpuid.h` file has it also in assembly. It is different shape than what we got, but fundamentally the same. It is trivial to write it from scratch using `asm` block. I do not think it even occurred to whoever was writing this to copy it from somewhere else.